dbt/tests/generic/test_is_null.sql

```diff
@@ -1,13 +1,23 @@
 -- Confirm that a column is null. Essentially, the opposite of the `not_null`
 -- test.
-{% test is_null(model, column_name, additional_select_columns=[]) %}
+{% test is_null(model, column_name, additional_select_columns=[], anti_join={}) %}
 
     select
         {% if additional_select_columns -%}
             {{ format_additional_select_columns(additional_select_columns) }},
         {%- endif %}
         {{ column_name }}
-    from {{ model }}
-    where {{ column_name }} is not null
+    from
+        (
+            select model.*
+            from (select * from {{ model }}) as model
+            {% if anti_join -%}
+                left join
+                    {{ anti_join.table_name }} as anti_join
+                    on model.{{ anti_join.join_on }} = anti_join.{{ anti_join.join_on }}
+                where anti_join.{{ anti_join.join_on }} is null
```

Comment:
We could imagine extending `join_on` to support arrays of values too, or dicts if we want to join on two columns that don't have the same name. For now, I'm keeping it as simple as possible, and I figure we can extend that in the future if we need to.